src/main/java/io/github/thebusybiscuit/slimefun4/implementation/listeners/PiglinListener.java

```diff
@@ -38,6 +56,29 @@ public void onEntityPickup(EntityPickupItemEvent e) {
         }
     }
 
+    @EventHandler
+    public void onPiglinDropItem(EntityDropItemEvent e) {
+        if (e.getEntity() instanceof Piglin) {
+            Set<ItemStack> drops = SlimefunPlugin.getRegistry().getBarterDrops(); 
+            
+            /*
+             * NOTE:  Getting a new random number each iteration because multiple items could have the same
+             * % chance to drop, and if one fails all items with that number will fail.  Getting a new random number
+             * will allow multiple items with the same % chance to drop.
+             */
+            
+            for (ItemStack is : drops) {
+                SlimefunItem sfi = SlimefunItem.getByItem(is);
+                if (sfi instanceof RandomMobDrop && ((RandomMobDrop)sfi).getMobDropChance() >= ThreadLocalRandom.current().nextInt(100)) {
+                    Item drop = e.getEntity().getWorld().dropItemNaturally(((Piglin)e.getEntity()).getEyeLocation(), sfi.getItem());
+                    drop.setVelocity(e.getItemDrop().getVelocity());
+                    e.getItemDrop().remove();
+                    return;
+                } 
+            }
+        }
```

Comment:
Honestly I would rather recommend making a seperate `PiglinBarterDrop` interface for this instead. (You can also just make it extend RandomMobDrop for simplicity).
Because this is a bit un-intuitive.